packages/xml-views-validation/src/validators/document/non-unique-id.ts

```diff
@@ -77,11 +73,7 @@ class IdsCollectorVisitor implements XMLAstVisitor {
       attrib.value !== "" &&
       attrib.syntax.value !== undefined &&
       attrib.parent.name !== null &&
-      // Heuristic to limit false positives by only checking tags starting with upper
-      // case names, This would **mostly** limit the checks for things that can actually be
-      // UI5 Elements / Controls.
-      /^[A-Z]/.test(attrib.parent.name) &&
-      !isNoneUI5id(attrib)
+      isPossibleCustomClass(attrib.parent as XMLElement & { name: string })
```

Comment:
why is casting needed, should not `attrib.parent.name !== null &&` cause narrowing of the type?